toqito/state_props/is_abs_ppt.py

```diff
@@ -0,0 +1,149 @@
+"""Checks if a quantum state is absolutely PPT (PPT from spectrum)."""
+
+import numpy as np
+
+from toqito.matrix_props import is_hermitian, is_positive_semidefinite, is_square
+
+
+def _abs_ppt_constraints(eigvals: np.ndarray, dim: list[int]) -> list[np.ndarray]:
+    r"""Generate the constraint matrices for the absolutely PPT test, following QETLAB/AbsPPTConstraints.
+
+    :cite:`Hildebrand_2007_PPT, QETLAB_link`.
+
+    Only implemented for 2x2, 2x3, and 3x3 systems.
+    """
+    eigvals = np.sort(np.real(eigvals))[::-1]
+    d1, d2 = dim
+    constraints = []
+
+    if (d1, d2) == (2, 2):
+        # Only one constraint: 2x2 matrix
+        # See Hildebrand, Eq. (2)
+        # Matrix: [[2*lam4, lam3-lam1], [lam3-lam1, 2*lam2]]
+        L = np.array(
+            [
+                [2 * eigvals[3], eigvals[2] - eigvals[0]],
+                [eigvals[2] - eigvals[0], 2 * eigvals[1]],
+            ]
+        )
+        constraints.append(L)
+    elif (d1, d2) in [(2, 3), (3, 2)]:
+        # Two constraints: 3x3 matrices
+        # See Hildebrand, Eq. (3)
+        # eigvals: lam1 >= ... >= lam6
+        L1 = np.array(
+            [
+                [2 * eigvals[5], eigvals[4] - eigvals[0], eigvals[3] - eigvals[1]],
+                [eigvals[4] - eigvals[0], 2 * eigvals[2], eigvals[1] - eigvals[3]],
+                [eigvals[3] - eigvals[1], eigvals[1] - eigvals[3], 2 * eigvals[0]],
+            ]
+        )
+        L2 = np.array(
+            [
+                [2 * eigvals[5], eigvals[4] - eigvals[0], eigvals[2] - eigvals[1]],
+                [eigvals[4] - eigvals[0], 2 * eigvals[3], eigvals[1] - eigvals[2]],
+                [eigvals[2] - eigvals[1], eigvals[1] - eigvals[2], 2 * eigvals[0]],
+            ]
+        )
+        constraints.extend([L1, L2])
+    elif (d1, d2) == (3, 3):
+        # Two constraints: 3x3 matrices
+        # See QETLAB/AbsPPTConstraints and Hildebrand, Eq. (4)
+        # eigvals: lam1 >= ... >= lam9
+        L1 = np.array(
+            [
+                [2 * eigvals[8], eigvals[7] - eigvals[0], eigvals[6] - eigvals[1]],
+                [eigvals[7] - eigvals[0], 2 * eigvals[5], eigvals[4] - eigvals[2]],
+                [eigvals[6] - eigvals[1], eigvals[4] - eigvals[2], 2 * eigvals[3]],
+            ]
+        )
+        L2 = np.array(
+            [
+                [2 * eigvals[8], eigvals[7] - eigvals[0], eigvals[5] - eigvals[1]],
+                [eigvals[7] - eigvals[0], 2 * eigvals[6], eigvals[4] - eigvals[2]],
+                [eigvals[5] - eigvals[1], eigvals[4] - eigvals[2], 2 * eigvals[3]],
+            ]
+        )
+        constraints.extend([L1, L2])
+    else:
+        # Not implemented for higher dimensions
+        raise NotImplementedError("Absolutely PPT constraints for dimensions > 3x3 are not implemented.")
```

Comment:
I believe we will want to implement the SDP that is also implemented in QETLAB here in this case, correct?
https://github.com/nathanieljohnston/QETLAB/blob/f589c35a29e03624ecfd972fd209014799d24862/IsAbsPPT.m#L95